Project-Aurora/Project-Aurora/Profiles/RocketLeague/Layers/RocketLeagueGoalExplosionLayerHandler.cs

```diff
@@ -0,0 +1,198 @@
+using System.Windows.Controls;
+
+namespace Aurora.Profiles.RocketLeague.Layers
+{
+    public class RocketLeagueGoalExplosionProperties : LayerHandlerProperties<RocketLeagueGoalExplosionProperties>
+    {
+        public bool? _ShowFriendlyGoalExplosion { get; set; }
+
+        [JsonIgnore]
+        public bool ShowFriendlyGoalExplosion { get { return Logic._ShowFriendlyGoalExplosion ?? _ShowFriendlyGoalExplosion ?? true; } }
+
+        public bool? _ShowEnemyGoalExplosion { get; set; }
+
+        [JsonIgnore]
+        public bool ShowEnemyGoalExplosion { get { return Logic._ShowEnemyGoalExplosion ?? _ShowEnemyGoalExplosion ?? true; } }
+
+        public bool? _Background { get; set; }
+
+        [JsonIgnore]
+        public bool Background { get { return Logic._Background ?? _Background ?? true; } }
+
+        public RocketLeagueGoalExplosionProperties() : base()
+        {
+
+        }
+
+        public RocketLeagueGoalExplosionProperties(bool arg = false) : base(arg)
+        {
+
+        }
+
+        public override void Default()
+        {
+            base.Default();
+            _PrimaryColor = Color.FromArgb(125,0,0,0);
+            _ShowEnemyGoalExplosion = true;
+            _ShowFriendlyGoalExplosion = true;
+            _Background = true;
+        } 
+    }
+
+    public class RocketLeagueGoalExplosionLayerHandler : LayerHandler<RocketLeagueGoalExplosionProperties>
+    {
+        private int previousOwnTeamGoals = 0;
+        private int previousOpponentGoals = 0;
+
+        private AnimationTrack[] tracks =
+        {
+            new AnimationTrack("Goal Explosion Track 0", 1.0f, 0.0f),
+            new AnimationTrack("Goal Explosion Track 1", 1.0f, 0.5f),
+            new AnimationTrack("Goal Explosion Track 2", 1.0f, 1.0f),
+            new AnimationTrack("Goal Explosion Track 3", 1.0f, 1.5f),
+            new AnimationTrack("Goal Explosion Track 4", 1.0f, 2.0f)
+        };
+
+        private long previoustime = 0;
+        private long currenttime = 0;
+
+        private static float goalEffect_keyframe = 0.0f;
+        private const float goalEffect_animationTime = 3.0f;
+
+        private bool showAnimation_Explosion = false;
+        public RocketLeagueGoalExplosionLayerHandler() : base()
+        {
+            _ID = "RocketLeagueGoalExplosion";
+        }
+
+        public override EffectLayer Render(IGameState gamestate)
+        {
+            previoustime = currenttime;
+            currenttime = Utils.Time.GetMillisecondsSinceEpoch();
+
+            EffectLayer layer = new EffectLayer("Goal Explosion");
+            AnimationMix goal_explosion_mix = new AnimationMix();
+
+            if (gamestate is GameState_RocketLeague)
+            {
+                var state = gamestate as GameState_RocketLeague;
+                var ownTeam = state.Player.Team == 0 ? state.Match.Blue : state.Match.Orange;
+                var opponentTeam = state.Player.Team == 1 ? state.Match.Blue : state.Match.Orange;
```

Comment:
Related to what I mentioned above, there should probably be some getter for getting 'Your Team' and the 'Opponent Team'.